modules/data-streams/src/main/java/org/elasticsearch/datastreams/DataStreamFeatures.java

```diff
@@ -37,7 +37,8 @@ public Set<NodeFeature> getFeatures() {
             DataStreamLifecycleHealthInfoPublisher.DSL_HEALTH_INFO_FEATURE,  // Added in 8.12
             LazyRolloverAction.DATA_STREAM_LAZY_ROLLOVER,                    // Added in 8.13
             DataStreamAutoShardingService.DATA_STREAM_AUTO_SHARDING_FEATURE,
-            DataStreamGlobalRetention.GLOBAL_RETENTION                       // Added in 8.14
+            DataStreamGlobalRetention.GLOBAL_RETENTION,                      // Added in 8.14
+            LazyRolloverAction.FAILURE_STORE_LAZY_ROLLOVER                   // Added in 8.15
```

Comment:
Do we need to add a node feature for now? We still have a way to limit rest test versions on the old version schemes until 8.16 is branched. If this feature wasn't behind a feature flag this would certainly be needed, but since it is do we need this?

I would propose that we use the feature flag check to guard the new logic instead of a node feature. When we remove the feature flag we can introduce a single node feature for the premiere of failure stores. Thoughts?

Reply:
I added the feature as per Mary's [suggestion](https://github.com/elastic/elasticsearch/pull/108108#discussion_r1599868090) - not for YAML REST tests, but for validation in [TransportRolloverAction.java](https://github.com/elastic/elasticsearch/pull/108108/files#diff-f165f6db589dbdcbde3400f8377579f9025c4173a192df11d35c9d8fd80ad703) and [BulkOperation.java](https://github.com/elastic/elasticsearch/pull/108108/files#diff-318211e7d42c8640831a65881e33020a927a2ce6342cf744bdad30ca755f9aed). Or are you suggesting to use the feature flag in those checks?

Reply:
I think we should just use the feature flag since we don't expect people to be doing rolling upgrades with feature flags enabled. 

We'll still need to guard against the test cases running for older versions. I think we should be able to set up something using the new capabilities API - preferably a test specific capability on the roll over rest API that we can remove/replace when we take the feature into tech preview.

Reply:
See [06c8b26](https://github.com/elastic/elasticsearch/pull/108108/commits/06c8b26546a87d3cb80ca543c3866e969b90a05e)